recipes/rocksdb/all/test_package/conanfile.py

```diff
@@ -1,22 +1,29 @@
-from conans import ConanFile, CMake, tools
-import os
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "CMakeDeps", "VirtualRunEnv"
+    test_type = "explicit"
+
+    def layout(self):
+        cmake_layout(self)
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["ROCKSDB_SHARED"] = self.dependencies["rocksdb"].options.shared
+        tc.generate()
 
     def build(self):
         cmake = CMake(self)
-        cmake.definitions["ROCKSDB_SHARED"] = self.options["rocksdb"].shared
         cmake.configure()
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self):
-            if not self.options["rocksdb"].shared:
-                bin_path = os.path.join("bin", "test_package_cpp")
-                self.run(bin_path, run_environment=True)
-
-            bin_path = os.path.join("bin", "test_package_stable_abi")
-            self.run(bin_path, run_environment=True)
+        if can_run(self):
+            self.run(f"ctest --output-on-failure -C {self.settings.build_type}", env="conanrun")
```

Comment:
- This is misleading for external contributors that might think that test_package should run library tests instead of package tests
- Is this change motivated just to avoid two calls to self.run? Usually logic in-recipe is better than in the CMakeLists.
- This has a downside of not printing the output if it passes, so information about the test run might be lost in our CI systems

All in all I'd say that it's better to keep the old logic of tests, they are standard Conan that everyone is used to it

Reply:
> Is this change motivated just to avoid two calls to self.run? Usually logic in-recipe is better than in the CMakeLists.

The motivation is always the same when you see ctest in test package, and it's not new: conan v2 doesn't offer access to dependencies options in test(), this is why we have now several ctest in test package when availability of an executable we want to run depends on dependency option.

It has already been reported to conan maintainers at the beginning of v2 migration, and ctest trick has been introduced by @jcar87, so I'm surprised to now see these reviews about ctest.